main/lsp/LSPConfiguration.h

```diff
@@ -122,6 +122,9 @@ class LSPConfiguration {
     std::unique_ptr<Location> loc2Location(const core::GlobalState &gs, core::Loc loc) const;
     bool isFileIgnored(std::string_view filePath) const;
 
+    // Returns true if this path has an extension that sorbet can handle.
+    bool hasAllowedExtension(std::string_view filePath) const;
+
```

Comment:
I considered folding this into `isFileIgnored`, but ultimately felt it was nicer to have the two separated. Happy to merge it back in if that seems like a more targeted fix.

Reply:
Can we at least use this new helper in common.cc where we read files from the disk?

https://github.com/sorbet/sorbet/compare/jez-allowed-extensions#diff-e3fe2cf8c033247f3ac88f1d0cd437714c54140e141c7b8084f6b445db6bbb7bR346

The way we got ourselves into this mess was by having LSP logic that differs from pipeline logic. I was hoping to take the opportunity in this PR to make them align.

This would mean moving this helper out of `main/lsp/` and into someplace more common.